meteor/client/member/checkIntoEvent.js

```diff
@@ -1,97 +1,159 @@
   } else {
-    Session.set('eventId', null);
+    
+    // No event selected
+    $('#searchDiv').show();
+    $('#checkIntoEventDiv').hide();
+
+    eventButtonToggle.set({
+      eventSelectText: 'Select',
+      eventSelectClass: 'btn-primary'
+    });
+
   }
+}
+
+// Runs the function every time the session var changes
+Tracker.autorun(function() {
+  setToggleValues();
+});
+
+// Sets the map markers
+// This is called in the helper function and is throttled for performance
+var setMapMarkers = _.debounce(function(eventsArray) {
+
+  if (eventsArray) {
+    gmaps.addMarkerCollection(eventsArray);
+    gmaps.calcBounds();
+  }
+
+}, 1000);
+
+// -----------------------------------------------------------------
+
+Template.checkIntoEvent.created = function () {
+  gmaps.initialize();
+};
+
+Template.checkIntoEvent.rendered = function() {
+  
+  // We don't want to start out with an event selected
+  Session.set('selectedEvent', null);
+  setToggleValues();
+
+  // Populate the event list on load with no filters
+  CheckinEventsSearch.search('');
+
 };
 
 Template.checkIntoEvent.helpers({
-  'geolocationSuccessful': function() {
-    return Session.get('longitude') && Session.get('eventId');
+  
+  'getEvents': function() {
+
+    var eventsArray = getEventsData();
+    setMapMarkers(eventsArray);    
+
+    return eventsArray;
+  },
+
+  'eventSelectText': function() {
+    return eventButtonToggle.get().eventSelectText;
   },
-  //TODO: make sure this is actually only active events
-  'currentEvents': function() {
-    return Events.currentEvents();
+
+  'eventSelectClass': function() {
+    return eventButtonToggle.get().eventSelectClass;
   },
-  //TODO: have a graceful "please wait" screen while geolocating
-  'eventSelected': function() {
-    return Session.get('eventId');
+
+  // Disables the accordion when there is a selected event
+  'dataToggle': function() {
+    return Session.get('selectedEvent') ? '' : 'collapse';
   }
+
 });
 
 Template.checkIntoEvent.events({
-  'change #eventSelector': function(e) {
-    e.preventDefault();
-    Session.set('eventId', $('#eventSelector option:selected').val());
-  },
-  'click #geoLocate': function(e) {
-    e.preventDefault();
-    
-    var geoOptions = {
-      enableHighAccuracy: true,
-      timeout: 5000,
-      maximumAge: 0
-    };
-
-    if(navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition(function(position) {
-
-
-        Session.set('longitude', position.coords.longitude);
-        Session.set('latitude', position.coords.latitude);
-          
-      }, function(error) {
-        addErrorMessage(error.message);
-      }, geoOptions);
-    } else {
-      addErrorMessage('Geolocation not supported');
-    }
+
+  // Automatically populates the search list on keyup
+  'keyup #eventSearchBox': _.throttle(function(e) {
+    CheckinEventsSearch.search($('#eventSearchBox').val().trim());
+  }, 200),
+
+  'click .in button': function(e) {
```

Comment:
@drewproud I don't understand this event...where is the .in button on checkIntoEvent.html?